syntax = "proto3";

service BookService {
    rpc getBookById (BookRequest) returns (BookResponse);
    rpc getAllBooks (EmptyRequest) returns (BookListResponse);
}

message BookRequest {
    int32 id = 1;
}

message BookResponse {
    int32 id = 1;
    string title = 2;
    string author = 3;
}

message EmptyRequest {}

message BookListResponse {
    repeated BookResponse books = 1;
}